presto-main/src/main/java/com/facebook/presto/execution/TaskManagerConfig.java

```diff
@@ -455,6 +456,19 @@ public TaskManagerConfig setHttpTimeoutThreads(int httpTimeoutThreads)
         return this;
     }
 
+    @Min(1)
+    public int getHttpTimeoutConcurrency()
+    {
+        return httpTimeoutConcurrency;
+    }
+
+    @Config("task.http-timeout-concurrency")
```

Comment:
Number of threads and concurrency is not totally clear because usually those are these same thing. What is the documentation update associated with this change? Can we make it clearer both in the code and in the docs how this config should be used?

Also the fact that division takes place here https://github.com/prestodb/airlift/blob/master/concurrent/src/main/java/com/facebook/airlift/concurrent/ConcurrentScheduledExecutor.java#L77 and you might get less threads than you asked for is not intuitive.

Ideally the config would be # of executors and # of threads per executor. I am guessing we did it this way to maintain compatibility with existing configuration?

Reply:
@aweisberg : Try to press "y" on the Github source page to get commit id attached to the link ;) . Otherwise the link might expire once the file changed

https://github.com/prestodb/airlift/blob/49d3830a5bee4804c91e3355d65c44267e4bde40/concurrent/src/main/java/com/facebook/airlift/concurrent/ConcurrentScheduledExecutor.java#L77

Reply:
I agree with @aweisberg . I also don't understand what's the difference between "concurrency" and "threads". 

Otherwise, the PR looks good :)

Reply:
I knew it was broken the moment I did it. And I was like... ahhh it will be right long enough for us to finish this. Thanks for the tip.

Reply:
@aweisberg @wenleix  The naming and calculation of threads per executor follows convention in https://github.com/prestodb/airlift/blob/master/http-client/src/main/java/com/facebook/airlift/http/client/HttpClientConfig.java#L545-L565

It is OK to have ``minimum threads per pool * pool number < total threads limit``.  Maximal threads in each pool is unchanged (default to Integer.MAX_VALUE).

For backward compatibility,  keep the concept of existing ``task.http-timeout-threads`` is desired . Users can optionally add new property ``task.http-timeout-concurrency`` for better scalability. 

Explanations of these properties are added inline.

Reply:
@viczhang861 : Thanks @viczhang861  for the explanation. LGTM.